test/FsAutoComplete.Tests.Lsp/ExtensionsTests.fs

```diff
@@ -375,3 +375,50 @@ let analyzerTests state =
         do! server.Shutdown()
       })
     ]
+
+let signatureTests state =
+  let server =
+    async {
+      let path = Path.Combine(__SOURCE_DIRECTORY__, "TestCases", "Tooltips")
+      let scriptPath = Path.Combine(path, "Script.fsx")
+      let! (server, events) = serverInitialize path defaultConfigDto state
+      do! waitForWorkspaceFinishedParsing events
+      do! server.TextDocumentDidOpen { TextDocument = loadDocument scriptPath }
+      match! waitForParseResultsForFile "Script.fsx" events with
+      | Ok () ->
+        () // all good, no parsing/checking errors
+      | Core.Result.Error errors ->
+        failtestf "Errors while parsing script %s: %A" scriptPath errors
+
+      return server, scriptPath
+    }
+    |> Async.Cache
+
+  let verifySignature line character expectedSignature =
+    testCaseAsync (sprintf "fsharp/signature for line %d character %d should be '%s'" line character expectedSignature) (async {
+      let! server, scriptPath = server
+      let pos: TextDocumentPositionParams = {
+        TextDocument =  { Uri = sprintf "file://%s" scriptPath }
+        Position = { Line = line; Character = character }
+      }
+      match! server.FSharpSignature pos with
+      | Ok { Content = content } ->
+        let r = JsonSerializer.readJson<CommandResponse.ResponseMsg<string>>(content)
+        Expect.equal r.Kind "typesig" "Should have a kind of 'typesig'"
+        Expect.equal r.Data expectedSignature (sprintf "Should have a signature of '%s'" expectedSignature)
+      | Result.Error errors ->
+        failtestf "Error while getting signature: %A" errors
+    })
+
+  testSequenced <|
+    testList "signature evaluation" [
+      testList "tests" [
+        verifySignature 0 4 "val arrayOfTuples : (int * int) []"  // verify that even first letter of the signature triggers correctly
```

Comment:
what I'd really love to be able to get to at some point with many of these tests is testing a range of similar positions start to finish, eg the entire length of an identifier. That would be the most durable test, but in lieu of that can you also add one that uses the end of the identifier as well? just to cover the extremes.

Reply:
Changed it so it tests all positions between a given start and end. I've kept it in a single testCaseAsync so a 12 character identiefer does not produce 12 separate tests.  If it fails it will still say at which character it failed first.